### PR TITLE
move middleware API section to api-reference page

### PR DESCRIPTION
**Pull request by sarah11918**
#### Description (required)

This separates out the API reference for Middleware from the guide page, and puts it on the API reference (where it should be! :smile: )

Content HAS been revised:
- Middleware guide content reordered for flow
- Some explanations for `content` and using `content.locals` to store data have been brought up into main guide description, removed from API reference (since they are not part of `astro:middleware` itself, only used by it)
- API reference is now exclusively stuff not elsewhere documented on the API page, and some code examples added

### CODE REVIEW

**Review by ematipico — APPROVED** [review]
The gut check passed!

File: src/content/docs/en/guides/integrations-guide/netlify.mdx [hunk 1]
```diff
@@ -92,7 +92,7 @@ To enable this, set the `edgeMiddleware` config option to `true`:
 
 Netlify Edge Functions provide a [context object](https://docs.netlify.com/edge-functions/api/#netlify-specific-context-object) including metadata about the request, such as a user’s IP, geolocation data, and cookies.
 
-To expose values from this context to your site, create a `netlify-edge-middleware.ts` (or `.js`) file in your project’s [source directory](/en/reference/configuration-reference/#srcdir). This file must export a function that returns the data to add to [Astro’s `locals` object](/en/guides/middleware/#locals), which is available in middleware and Astro routes.
+To expose values from this context to your site, create a `netlify-edge-middleware.ts` (or `.js`) file in your project’s [source directory](/en/reference/configuration-reference/#srcdir). This file must export a function that returns the data to add to [Astro’s `locals` object](/en/reference/api-reference/#locals), which is available in middleware and Astro routes.
```
Comment (sarah11918) [inline src/content/docs/en/guides/integrations-guide/netlify.mdx:95]:
```suggestion
To expose values from this context to your site, create a `netlify-edge-middleware.ts` (or `.js`) file in your project’s [source directory](/en/reference/configuration-reference/#srcdir). This file must export a function that returns the data to add to [Astro’s `locals` object](/en/guides/middleware/#storing-data-in-contextlocals), which is available in middleware and Astro routes.
```

File: src/content/docs/en/guides/integrations-guide/netlify.mdx [hunk 2]
```diff
@@ -164,7 +164,7 @@ You can still include a `public/_redirects` file for manual redirects. Any redir
 
 [Netlify On-demand Builders](https://docs.netlify.com/configure-builds/on-demand-builders/) are serverless functions used to generate web content as needed that’s automatically cached on Netlify’s Edge CDN. You can enable these functions using the [`builders` configuration](#builders).
 
-By default, all pages will be rendered on first visit and the rendered result will be reused for every subsequent visit until you redeploy. To set a revalidation time, call the [`runtime.setBuildersTtl(ttl)` local](/en/guides/middleware/#locals) with the duration (in seconds).
+By default, all pages will be rendered on first visit and the rendered result will be reused for every subsequent visit until you redeploy. To set a revalidation time, call the [`runtime.setBuildersTtl(ttl)` local](/en/reference/api-reference/#locals) with the duration (in seconds).
```
Comment (sarah11918) [inline src/content/docs/en/guides/integrations-guide/netlify.mdx:167]:
```suggestion
By default, all pages will be rendered on first visit and the rendered result will be reused for every subsequent visit until you redeploy. To set a revalidation time, call the [`runtime.setBuildersTtl(ttl)` local](/en/reference/api-reference/#astrolocals) with the duration (in seconds).
```

File: src/content/docs/en/reference/adapter-reference.mdx [hunk 3]
```diff
@@ -201,7 +201,7 @@ This method calls the Astro page that matches the request, renders it, and retur
 const response = await app.render(request);
 ```
 
-The method accepts a mandatory `request` argument, and two other optional arguments: [`routeData`](/en/reference/integrations-reference/#routedata-type-reference) and [`locals`](/en/guides/middleware/#locals).
+The method accepts a mandatory `request` argument, and two other optional arguments: [`routeData`](/en/reference/integrations-reference/#routedata-type-reference) and [`locals`](/en/reference/api-reference/#locals).
```
Comment (sarah11918) [inline src/content/docs/en/reference/adapter-reference.mdx:204]:
```suggestion
The method accepts a mandatory `request` argument, and two other optional arguments: [`routeData`](/en/reference/integrations-reference/#routedata-type-reference) and [`locals`](/en/reference/api-reference/#astrolocals).
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 4]
```diff
@@ -1133,6 +1133,53 @@ const blog = defineCollection({
...
+
+<p>
+  <Since v="2.6.0" />
+</p>
+
+Middleware allows you to intercept requests and responses and inject behaviors dynamically every time a page or endpoint is about to be rendered. For features and usage examples, [see our middleware guide](/en/guides/middleware/).
+
+### `onRequest()`
+
+A required exported function from `src/middleware.js` that will be called before rendering every page or API route. It accepts two optional arguments: [context](#localscontext) and [next()](#next). `onRequest()` must return a `Response`: either directly, or by calling `next()`.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:1146]:
```suggestion
A required exported function from `src/middleware.js` that will be called before rendering every page or API route. It accepts two optional arguments: [context](#contextlocals) and [next()](#next). `onRequest()` must return a `Response`: either directly, or by calling `next()`.
```

**Review by dreyfus92 — APPROVED** [review]
LGTM 🫡

**Pull request merged**